<launch>

    <!-- Example - Agruments -->
    <arg name="example_string_param" default="example"/>
    <arg name="example_double_param" default="1"/>
    <arg name="example_bool_param"   default="true"/>

    <!-- Global Parameters -->
    <param name="example_string_param" value="$(arg example_string_param)"/>
    <param name="example_double_param" value="$(arg example_double_param)"/>
    <rosparam command="load" file="$(find example_package)/config/example.yaml"/>

    <!-- Launch C++ Nodes -->
    <node pkg="example_package" type="example_cpp_node" name="example_cpp_node" output="screen">
        <param name="example_bool_param" value="false"/>
        <rosparam param="example_vector_param">[0.0, 0.0]</rosparam>
    </node>

    <!-- Launch Python Nodes -->
    <node pkg="example_package" type="example.py" name="example_python_node" output="screen">
        <param name="example_bool_param" value="false"/>
        <rosparam param="example_vector_param">[0.0, 0.0]</rosparam>
    </node>

    <!-- Include Other Launchfiles -->
    <include ns="namespace_1" file="$(find example_package)/launch/dummy_launchfile.launch">
        <arg name="example_string_param" value="new_example"/>
        <arg name="example_double_param" value="$(arg example_double_param)"/>
    </include>

</launch>
